load.go

```diff
@@ -361,7 +361,15 @@ func (s *Spec) UnmarshalYAML(dt []byte) error {
 
 	// Remove extension nodes from the main AST and store them so we can unmarshal
 	// them separately.
-	body := parsed.Docs[0].Body.(*ast.MappingNode)
+	b := parsed.Docs[0].Body
```

Comment:
Did this case come up because of a change in the yaml library?

Reply:
Yes, it caused a panic in tests.